tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/util/StringUtil.java

```diff
@@ -0,0 +1,108 @@
+package com.google.cloud.bigquery.utils.queryfixer.util;
+
+import lombok.Value;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/** A utility class to provide static helper methods regarding String. */
+public class StringUtil {
+
+  /**
+   * Find word(s) from a dictionary that are most similar to a target word. The similarity is
+   * measured by edit distance between two words.
+   *
+   * @param dict dictionary of words
+   * @param target target word
+   * @return a list of Strings and their edit distance to the target.
+   */
+  public static SimilarStrings findSimilarWords(Collection<String> dict, String target) {
+    List<Pair<Integer, String>> distanceWordPairs =
+        dict.stream()
+            .map(word -> Pair.of(editDistance(word, target), word))
+            .collect(Collectors.toList());
+
+    if (distanceWordPairs.isEmpty()) {
+      return SimilarStrings.empty();
+    }
+
+    Integer minDistance =
+        distanceWordPairs.stream().min(Comparator.comparingInt(Pair::getLeft)).get().getLeft();
+
+    List<String> words =
+        distanceWordPairs.stream()
+            .filter(pair -> (pair.getLeft().equals(minDistance)))
+            .map(Pair::getRight)
+            .collect(Collectors.toList());
+    return new SimilarStrings(words, minDistance);
+  }
+
+  // the code was inspired from
+  // https://www.programcreek.com/2013/12/edit-distance-in-java/
```

Comment:
I think this is very common algorithm you don't have to comment on this :)

Reply:
This comment was removed.